vmm/src/config.rs

```diff
@@ -0,0 +1,248 @@
+// Copyright © 2019 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use linux_loader::cmdline::Cmdline;
+use net_util::MacAddr;
+use std::net::Ipv4Addr;
+use std::path::Path;
+use std::result;
+use vm_memory::{GuestAddress, GuestUsize};
+
+pub const DEFAULT_VCPUS: u8 = 1;
+pub const DEFAULT_MEMORY: GuestUsize = 512;
+pub const DEFAULT_RNG_SOURCE: &str = "/dev/urandom";
+const CMDLINE_OFFSET: GuestAddress = GuestAddress(0x20000);
+
+/// Errors associated with VM configuration parameters.
+#[derive(Debug)]
+pub enum Error {
+    /// Failed parsing cpus parameters.
+    ParseCpusParams,
+    /// Failed parsing memory parameters.
+    ParseMemoryParams,
+    /// Failed parsing kernel parameters.
+    ParseKernelParams,
+    /// Failed parsing kernel command line parameters.
+    ParseCmdlineParams,
+    /// Failed parsing disk parameters.
+    ParseDiskParams,
+    /// Failed parsing random number generator parameters.
+    ParseRngParams,
+    /// Failed parsing network parameters.
+    ParseNetParams,
+}
+pub type Result<T> = result::Result<T, Error>;
+
+pub struct VmParams<'a> {
+    pub cpus: &'a str,
+    pub memory: &'a str,
+    pub kernel: &'a str,
+    pub cmdline: Option<&'a str>,
+    pub disk: Vec<&'a str>,
+    pub rng: Option<&'a str>,
+    pub net: Option<&'a str>,
+}
+
+pub struct CpusConfig {
+    num: u8,
+}
+
+impl CpusConfig {
+    pub fn new(cpus: &str) -> Result<Self> {
```

Comment:
Now i'm starting to drive this programatically it feels weird to turn by int into a string (from my testing). I think that it should be done by implementing From for both the underlying type and the string. But you're actually failing to report an error here if the user doesn't give an integral value...

Reply:
Fixed by making public every structures